packages/components/src/local/asset-icon/index.tsx

```diff
@@ -36,17 +36,24 @@ export const AssetIcon: React.FC<PropTypes> = ({
   })
 
   const clickEvent = (): void => {
-    setSelectedAsset({
-      uniqid,
-      name,
-      type,
-      force,
-      visibleTo,
-      controlledBy,
-      condition,
-      status
-    })
-    setShowMapBar(true)
+    if (selectedAsset && selectedAsset.uniqid === uniqid) {
+      // ok, it's the current asset. clear it
```

Comment:
Remember that I said comments shouldn't be conversational. This should be.

// Clear the current asset.